src/modules/Dropdown/Dropdown.js

```diff
@@ -811,7 +818,20 @@ export default class Dropdown extends Component {
     // filter by search query
     if (search && searchQuery) {
       if (_.isFunction(search)) {
-        filteredOptions = search(filteredOptions, searchQuery)
+        // poor man's memoization
+        // if the value, options, and search function are referentially equal to the last call
+        // just return the last value
+        const { lastValue, lastOptions, lastSearch } = this.getMenuOptions
+
+        if (lastValue === value && lastOptions === options && lastSearch === search) {
+          filteredOptions = this.getMenuOptions.lastReturn
+        } else {
+          this.getMenuOptions.lastValue = value
+          this.getMenuOptions.lastOptions = options
+          this.getMenuOptions.lastSearch = search
+
+          filteredOptions = search(filteredOptions, searchQuery)
+        }
```

Comment:
This seems unwise to add more state management, especially outside of this.state.  I thought my PR addressed this issue quite well by ensuring this.state stayed in sync with prop changes.  Did you have a chance to review my PR?